src/psyclone/tests/test_files/dynamo0p3/14.11_halo_required_clean_multi.f90

```diff
@@ -42,23 +43,26 @@ program halo_different_stencils
   ! stencil_cross stays as stencil_cross (which would happen in this
   ! example). However, the halo exchange library does make use of this
   ! information at the moment in any case.
-  use testkern_stencil_w3_mod, only: testkern_stencil_w3_type
+  use constants_mod,                  only: i_def
+  use field_mod,                      only: field_type
+  use flux_direction_mod,             only: y_direction
+  use testkern_stencil_w3_mod,        only: testkern_stencil_w3_type
   use testkern_stencil_xory1d_w3_mod, only: testkern_stencil_xory1d_w3_type
-  use inf,      only: field_type
-  use flux_direction_mod, only: y_direction
+
   implicit none
+
   type(field_type) :: f1, f2, f3
-  integer :: f2_extent=2
-  integer :: f2_direction=y_direction
+  integer(i_def)   :: f2_extent = 2
+  integer(i_def)   :: f2_direction = y_direction
 
-  call invoke(                                                       &
-       setval_c(f2, 0.0),                                            &
+  call invoke(                                                  &
+       setval_c(f2, 0.0),                                       &
        ! f1 is w3 and is written to
        ! f2 is w2 and is read with stencil cross
-       testkern_stencil_w3_type(f1,f2,f2_extent),                    &
+       testkern_stencil_w3_type(f1, f2, f2_extent),             &
        ! f3 is w3 and is written to
        ! f2 is w2 and is read with stencil xory1d
-       testkern_stencil_xory1d_w3_type(f3,f2,2,f2_direction) &
+       testkern_stencil_xory1d_w3_type(f3, f2, 2, f2_direction) &
```

Comment:
I wanted to do that originally but I was worried if it would be too "LFRic"-ky for something not necessarily being LFRic code. Done.

Reply:
Right, specifying kind for literal stencil depth exposes what I think is bug in PSyclone when distributed memory is used and when the literal stencil depth is provided by the algorithm layer. The actual error is `ValueError: invalid literal for int() with base 10: '2_i_def'`. The offending line is `self._literal_depth += int(value_str)` in `HaloReadAccess._compute_from_field` method in LFRic API. I reverted the example to just use "2". What are we going to do about this? Create a new issue and put "TODOs" in API and relevant Fortran examples?

Reply:
Oh yes. This will be fixed by #753 and therefore I don't think it's worth worrying about separately. Please just add a comment mentioning the problem and referencing #753.

Reply:
Added the reference in `dynamo0p3.py`, this kernel and other affected kernels with integer literal stencil depth modified in this PR.